File: src/main/java/edu/marmara/mapper/impl/StudentMapperImpl.java
package edu.marmara.mapper.impl;

import edu.marmara.dto.StudentGetDTO;
import edu.marmara.mapper.StudentMapper;
import edu.marmara.model.Advisor;
import edu.marmara.model.Course;
import edu.marmara.model.School;
import edu.marmara.model.Student;
import edu.marmara.model.Transcript;
import edu.marmara.repository.InstructorRepository;
import edu.marmara.repository.impl.InstructorRepositoryImpl;

import java.util.ArrayList;
import java.util.HashMap;

public class StudentMapperImpl implements StudentMapper {
    School school = School.getInstance();
    InstructorRepository instructorRepository = new InstructorRepositoryImpl();

    @Override
    public Student mapTo(StudentGetDTO studentGetDTO) {
        Student student = new Student();

        student.setUuid(studentGetDTO.getUuid());
        student.setName(studentGetDTO.getName());
        student.setSurname(studentGetDTO.getSurname());
        student.setEmail(studentGetDTO.getEmail());
        student.setBirthDate(studentGetDTO.getBirthDate());
        student.setStudentId(studentGetDTO.getStudentId());
        student.setYearEnrolled(studentGetDTO.getYearEnrolled());
        student.setSemester(studentGetDTO.getSemester());

        student.setTranscript(new Transcript());
        student.getTranscript().setPassedCourses(new HashMap<>());
        student.getTranscript().setFailedCourses(new ArrayList<>());
        student.getTranscript().setFailedCredit(0);
        student.getTranscript().setPassedCredit(0);
        student.getTranscript().setNotTakenCourses(new ArrayList<>());


        Advisor advisor = instructorRepository.findAdvisorByEmail(studentGetDTO.getAdvisor());
        if (advisor != null) {
            student.setAdvisor(advisor);
            advisor.getStudents().add(student);
        }

        return student;
    }
}

File: src/python/src/main/marmara/mapper/impl/StudentMapperImpl.py

from src.main.marmara.dto.StudentGetDTo import StudentGetDTO
from src.main.marmara.model import Student
from src.main.marmara.model import Transcript
from src.main.marmara.model import Schedule
from src.main.marmara.repository.impl import InstructorRepositoryImpl
class StudentMapperImpl:
    def mapTo(self, studentGetDTO: StudentGetDTO) -> Student:
        student = Student()

        student.uuid = studentGetDTO.uuid
        student.name = studentGetDTO.name
        student.surname = studentGetDTO.surname
        student.email = studentGetDTO.email
        student.birth_date = studentGetDTO.birth_date
        student.student_id = studentGetDTO.student_id
        student.year_enrolled = studentGetDTO.year_enrolled
        student.semester = studentGetDTO.semester

        student.transcript = Transcript()
        student.transcript.passed_courses = {}
        student.transcript.failed_courses = []
        student.transcript.failed_credit = 0
        student.transcript.passed_credit = 0
        student.transcript.not_taken_courses = []
        student.transcript.currently_taken_courses = []

        student.weekly_schedule = Schedule()
        student.weekly_schedule.courses = []

        instructor_repository = InstructorRepositoryImpl()
        advisor = instructor_repository.find_advisor_by_email(studentGetDTO.advisor)
        if advisor is not None:
            student.advisor = advisor
            advisor.add_student(student)

        return student
